### PR TITLE
fix: surface openSSL unhandled rejection errors that occur within plugin code

### PR DESCRIPTION
**Pull request by emilyrohrbough**
Pulls forward the changes made in https://github.com/cypress-io/cypress/pull/19572

- Related to #18914 & #19320

While investigating the above issues, it was asked why the OpenSSL error message wasn't up-front for consumers to reference.

### CODE REVIEW

File: packages/server/test/unit/plugins/child/run_require_async_child_spec.js [hunk 1]
```diff
@@ -34,6 +34,9 @@ describe('lib/plugins/child/run_require_async_child', () => {
 
       expect(tsNodeUtil.register).to.be.calledOnce
     })
+
+    // FIXME: need to validate that TS is checked once when ts is not found as well
+    it.skip('checks for typescript only once if typescript module was not found')
```
Comment (BlueWinds) [inline packages/server/test/unit/plugins/child/run_require_async_child_spec.js:39]: Is this change intentional? Not sure how it relates to the main purpose of the PR.

Comment (emilyrohrbough) [reply packages/server/test/unit/plugins/child/run_require_async_child_spec.js:39]: Theres quite a few server/plugin/child tests that are currently skipped right now. This was a test-case I noticed that wasn't tested. Yes intentional to add & you are correct is doesn't necessarily relate to the PR changes.

**Review by BlueWinds — APPROVED** [review]

**Review by mjhenkes — APPROVED** [review]
